articles/how-to-use-graphical-tiles.md

```diff
@@ -0,0 +1,54 @@
+# How to use Graphical Tiles
+
+![screenshot of sadconsole with graphical tiles](~/images/graphical-tiles.png)
+
+SadConsole supports graphical tiles! It's recommended that you have at least something basic drawn on-screen (eg. character, floor, walls) before adding graphics, but you can do this at any stage of development. As you can see from the screenshot, you can mix graphical tiles with regular fonts.
+
+## How Graphical Tiles Work
+
+If you read the [basic font information](~/articles/basic-font-information.md) tutorial, you'll see that SadConsole uses simple PNG images for fonts. We can leverage this by creating a font image that uses regular graphical tiles as a spritesheet, instead of ASCII characters.
+
+When you substitue graphics, pay attention to the position in the font file. With an ASCII font, writing the `@` glyph renders the `@` image; in the spriesheet, it'll render whatever image occupies that same cell in the spritesheet.
+
+## Creating the Font Spritesheet
+
+To start out, procure or create the images you need. You can also use the excellent [1-bit Pack](https://kenney.nl/assets/bit-pack) by Kenney, which includes numerous high-quality sprites.
+
+Next, create an empty spritesheet in your graphical editing software of choice. Ideally, include a grid, so you can easily distinguish tiles. Draw or copy/paste a few characters into the appropriate position. If you're not sure which cell corresponds to which character, you can refer to the existing font files, such as the `IBM_8x16.png` file.
+
+Hide the grid and export the file into your project, under the `Fonts` directory. Next, create the `.font` metadata file. Here's a sample:
+
+```
+{
+    "FilePath": "MyCustomFont_16x16.png",
+    "GlyphHeight": 16,
+    "GlyphPadding": 0,
+    "GlyphWidth": 16,
+    "Name": "MyCustomFont",
+    "SolidGlyphIndex": 219
+}
+```
+
+Replace the font sizes and the filename with appropriate values. Note the `SolidGlyphIndex` value: it's used whenever you print an empty tile, or call a method to fill a console.
+
+## Using the Spritesheet Font In-Game
+
+Somewhere in your game console constructor, set the font:
+
+```
+var fontMaster = SadConsole.Global.LoadFont("Fonts/MyCustomFont.font");
+var normalSizedFont = fontMaster.GetFont(SadConsole.Font.FontSizes.One);
+this.Font = normalSizedFont;
+```

```

Comment:
This can be improved. If you keep loading the font over and over you end up with different texture objects for console's that have already loaded a font. For example.. If you have this in your console constructor and use something like

```csharp
var console1 = new MyConsole();
var console2 = new MyConsole();
```

Each console (in the constructor) is loading a new copy of the font. This means that two textures have been loaded into the graphics card. You should instead load it one time at the start of your game, and then reference it from then on.

```csharp
private static void Init()
{
    SadConsole.Global.LoadFont("Fonts/MyCustomFont.font");

    var console = new SadConsole.Console(10, 10, SadConsole.Global.Fonts["MyCustomFont"].GetFont(Font.FontSizes.One));
}
```

This loads the font into the system and caches it in the `Global.Fonts` collection. The `Name` setting from the font definition file is the id used to reference the font.

If you want your own console type to **always** use the custom font, you should modify the constructor to reference it, since it was loaded at startup

```csharp
private class MyConsole : Console
{
    public MyConsole(int width, int height) : base(width, height, SadConsole.Global.Fonts["MyCustomFont"].GetFont(Font.FontSizes.One))
    {

    }
}
```

Reply:
This is interesting. My game is one giant console and I never got sub-consoles to work properly (had to end up coding my own work-arounds, which have their own bugs).

I'll update my article to mention doing this in the game constructor and just using the global font by reference. I think that's the most elegant solution.